### PR TITLE
add bullet gem, fix some of our identified n+1 joins

### LINKED ISSUE #2225: Fix postgres n+1 joins
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**

For expedience, we're adding some n+1 joins to some controllers. We shouldn't let that sit for too long though!

* **What feature or behavior is this required for?**

fast app

* **How could we solve this issue? (Not knowing is okay!)**

`git grep "n+1 join here"` to find these trouble spots; then adjust the controller code to use appropriate `includes` or `joins` to eliminate the n+1 joins.

* **Anything else?**

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

This cropped up again in an inconvenient way, so going to make this my 'chip away at it' project this week.

This pull request makes the following changes:
* install bullet gem
* make bullet fail tests if it hits a problem
* fix the problems

It relates to the following issue #s: 
* bumps #2225 

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: app/models/concerns/exportable.rb [hunk 1]
```diff
@@ -127,11 +127,11 @@ def last_call_timestamp
   end
 
   def call_count
-    calls.count
+    calls.size
```
Comment (colinxfleming) [inline app/models/concerns/exportable.rb:130]: Fun fact: `size` will use the cached query, and `count` will always run a `count(*)` query. So this winds up saving us a lot of queries.

File: app/models/concerns/exportable.rb [hunk 2]
```diff
@@ -158,17 +158,11 @@ def get_field_value_for_serialization(field)
...
 
   def external_pledge_sum
-    sum = 0
-    all_pledges = external_pledges.all
-
-    all_pledges.each do |pledge|
-      sum += pledge.try :amount
-    end
-    sum
+    external_pledges.inject { |x, acc = 0| acc = x.try(:amount) || 0}
```
Comment (colinxfleming) [inline app/models/concerns/exportable.rb:165]: `external_pledges.all` was running an additional select * from external pledges, despite eager loading. So we turn this into a snappy one line reducer.

**Review by colinxfleming — COMMENTED** [review]
notes

File: app/models/concerns/call_listable.rb [hunk 3]
```diff
@@ -69,7 +69,7 @@ def clear_call_list(line)
 
   def ordered_patients(line)
     # n+1 join here
```
Comment (colinxfleming) [inline app/models/concerns/call_listable.rb:71]: There's more to do here, but we do fix some of the particular egregious cases (and get bullet to hush, so leaving this in.

Comment (lomky) [reply app/models/concerns/call_listable.rb:71]: since we've silence the alert, should we throw an issue on the backlog so we don't lose site of it?

**Review by lomky — APPROVED** [review]
_so freaking cool_. I love rails

File: app/models/concerns/call_listable.rb (hunk 3 above)
Comment (colinxfleming) [reply app/models/concerns/call_listable.rb:71]: I think let's keep #2225 open? I'll add a note about what remaining problems there are.

**Comment by colinxfleming** [comment]
thanks @lomky !

**Pull request merged**